### PR TITLE
PR: Show time units in Profiler

### LINKED ISSUE #3672: Profiler is not showing time units
As it is mentioned in this Stackoverflow post

http://stackoverflow.com/q/40509525/438386

our profiler is not showing the time units spent in every function or method call (e.g. seconds, minutes, etc). We really need to add them to make things clearer for our users.

@malliwi88, please work on this one.

### PR DESCRIPTION
**Pull request by malliwi88**
Fixes #3672.

----

Maybe it is necessary to do a little cleaning to the profiler, I found a lot of code not functional inside profiler.

The final result looks like this
spyder 3 1 0 dev0 python 3 5 _001

### CODE REVIEW

**Comment by malliwi88** [comment]
@ccordoba12 

I waiting for test for check if conflicts are solved now

**Comment by goanpeca** [comment]
@ccordoba12 is this ready?

File: spyder_profiler/widgets/profilergui.py [hunk 1]
```diff
@@ -510,25 +510,70 @@ def color_string(self, args):
                 color = "red"
         return [format[0] % x[0], [diff_str, color]]
 
+    def format_measure(self, data):
+
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Please add a docstring here

File: spyder_profiler/widgets/profilergui.py [hunk 2]
```diff
@@ -510,25 +510,70 @@ def color_string(self, args):
                 color = "red"
         return [format[0] % x[0], [diff_str, color]]
 
+    def format_measure(self, data):
+
+        formated_data = [0] * len(data)
+
+        for (i, measure) in enumerate(data):
+
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Remove this blank space

File: spyder_profiler/widgets/profilergui.py [hunk 3]
```diff
@@ -510,25 +510,70 @@ def color_string(self, args):
...
+    def format_output(self, child_key):
         """ Formats the data"""
+
         if True:
-            data = [x.stats.get(child_key,[0,0,0,0,0]) for x in self.stats1]
-            return map(self.color_string,zip(list(zip(*data))[1:4], [["%i"]*2, ["%.3f","%.3f"], ["%.3f","%.3f"]]))
+            data = [x.stats.get(child_key, [0,0,0,0,0]) for x in self.stats1]
+            # print(self.stats[child_key], list(zip(*data))[1:4])
+            #return map(self.color_string,
+            #           zip(list(zip(*data))[1:4], [["%i n"] * 2, ["%.3f ms", "%.3f ms"], ["%.3f", "%.3f"]]))
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Remove these commented lines if they are not necessary anymore

File: spyder_profiler/widgets/profilergui.py [hunk 4]
```diff
@@ -510,25 +510,70 @@ def color_string(self, args):
...
              ) = self.function_info(child_key)
+            """
+             ((total_calls, total_calls_dif), (loc_time, loc_time_dif), (cum_time,
+              cum_time_dif)) = self.format_output(child_key)
 
-            ((total_calls, total_calls_dif), (loc_time, loc_time_dif), (cum_time,
-             cum_time_dif)) = self.format_output(child_key)
+             (primcalls, total_calls, loc_time, cum_time, callers
+              ) = self.stats[child_key]
+             """
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Remove this block of commented code if it's not necessary anymore.

File: spyder_profiler/widgets/profilergui.py [hunk 5]
```diff
@@ -538,34 +583,36 @@ def populate_tree(self, parentItem, children_list):
...
+            child_item.setTextAlignment(2, Qt.AlignLeft)
+
+            child_item.setData(4, Qt.DisplayRole, loc_time_dif[0])
+            child_item.setForeground(4, QColor(loc_time_dif[1]))
+            child_item.setTextAlignment(4, Qt.AlignLeft)
+
             child_item.setData(6, Qt.DisplayRole, total_calls_dif[0])
             child_item.setForeground(6, QColor(total_calls_dif[1]))
             child_item.setTextAlignment(6, Qt.AlignLeft)
+            """
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Again, remove this if it's not necessary anymore.

**Comment by ccordoba12** [comment]
No, it's not. There's a lot of commented (and it's seems unnecessary code) that @malliwi88 needs to decide if he wants to leave or remove.

**Comment by malliwi88** [comment]
Ok @ccordoba12 your suggestions have been added and the test has passed :+1,

File: spyder_profiler/widgets/profilergui.py [hunk 6]
```diff
@@ -510,12 +510,50 @@ def color_string(self, args):
                 color = "red"
         return [format[0] % x[0], [diff_str, color]]
 
+    def format_measure(self, data):
+        """getting format and units for data coming from profiler task."""
+        formated_data = [0] * len(data)
+
+        for (i, measure) in enumerate(data):
+
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Remove this blank line

**Comment by malliwi88** [comment]
@ccordoba12 #3672 and #3735, they are solved

Now profiler is showing time units. I think that maybe it is necessary to do a little cleaning to the profiler, I found a lot of code not functional inside profiler.

At the beginning the profiler looks like this:
profiler_before

The final result looks like this:
spyder 3 1 0 dev0 python 3 5 _001

Any suggestions or corrections are welcome.

**Comment by goanpeca** [comment]
@malliwi88 just focus on the problem at hand. The PR should solve the immediate and specific problem not refactor a bunch of code.

File: spyder_profiler/widgets/profilergui.py [hunk 7]
```diff
@@ -538,34 +565,22 @@ def populate_tree(self, parentItem, children_list):
...
-                                      '(not in sub-functions)'))
+            child_item.setToolTip(3, _('Local time in function ' \
+                                       '(not in sub-functions)'))
 
             child_item.setData(3, Qt.DisplayRole, loc_time)
             child_item.setTextAlignment(3, Qt.AlignRight)
-            
-            child_item.setData(4, Qt.DisplayRole, loc_time_dif[0])
-            child_item.setForeground(4, QColor(loc_time_dif[1]))
-            child_item.setTextAlignment(4, Qt.AlignLeft)
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: Also this?

File: spyder_profiler/widgets/profilergui.py [hunk 8]
```diff
@@ -538,34 +565,22 @@ def populate_tree(self, parentItem, children_list):
...
-            child_item.setToolTip(5, _('Total number of calls '\
+            child_item.setToolTip(5, _('Total number of calls ' \
                                        '(including recursion)'))
 
             child_item.setData(5, Qt.DisplayRole, total_calls)
             child_item.setTextAlignment(5, Qt.AlignRight)
-            
-            child_item.setData(6, Qt.DisplayRole, total_calls_dif[0])
-            child_item.setForeground(6, QColor(total_calls_dif[1]))
-            child_item.setTextAlignment(6, Qt.AlignLeft)
```
Comment (ccordoba12) [inline spyder_profiler/widgets/profilergui.py]: And this?

**Comment by malliwi88** [comment]
@ccordoba12 ready :+1:

**Pull request merged**